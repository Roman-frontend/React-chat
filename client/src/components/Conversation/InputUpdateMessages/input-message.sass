.MuiInputBase-input
  min-height: 2rem !important