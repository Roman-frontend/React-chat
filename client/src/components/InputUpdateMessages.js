import React, {useEffect} from 'react'
import {useAuthContext} from '../context/AuthContext'
import {useMessagesContext} from '../context/MessagesContext'
import {useServer} from '../hooks/Server'

export default function InputUpdateMessages(props) {

  const {name, userId} = useAuthContext()
  const {messages, setMessages, messageActions, setMessageActions, inputRef} = useMessagesContext()
  const {postData, putData, getData} = useServer()
  const copyMessages = messages.slice(0, messages.length);
  let updatedArrayMessages = []

  useEffect(() => {
    inputRef.current.focus();
    getData()
  }, [ , userId]);

  function inputUpdateMessages(event) {
    if ((event.key === "Enter") && !(inputRef.current.value === "")) {
      if (messageActions.change) changeMessageText()
      else if (messageActions.answerTo) messageInReply(inputRef.current.value)
      else newMessage(inputRef.current.value)
      
      setMessages(updatedArrayMessages)   
      inputRef.current.value = null
    }
  }

  async function changeMessageText() {
    let putMessage = []

    updatedArrayMessages = messages.map(message => {
      if (message._id === messageActions.change) {
        message.text = inputRef.current.value
        putMessage.push(message)
        return message
      } else return message
    })
    
    await putData(putMessage[0], messageActions.change)
    const object = Object.assign({}, {...messageActions}, {'change': null})
    setMessageActions({...object})
  }

  function messageInReply(response) {
    const answerTo = messages.find(message => message._id === messageActions.answerTo);
    copyMessages.unshift({
      userId: userId,
      username: name, 
      text: answerTo.text, 
      createdAt: new Date().toLocaleString(), 
      _id: Date.now(), 
      reply: response
    },) 
 
    const object = Object.assign({}, {...messageActions}, {answerTo: null})
    setMessageActions({...object})    
    postData(`/api/chat/post-message`, copyMessages[0])
  }

  function newMessage(textMessage) {
    
    copyMessages.unshift({
      userId: userId,
      username: name, 
      text: textMessage, 
      createdAt: new Date().toLocaleString(), 
    },)  

    updatedArrayMessages = copyMessages
    postData(`/api/chat/post-message`, updatedArrayMessages[0])
  }

  return (
    <input 
      type="text" 
      className="input-message" 
      placeholder="Enter Text" 
      ref={inputRef} 
      onKeyUp={event => inputUpdateMessages(event)}
    />
  )
}