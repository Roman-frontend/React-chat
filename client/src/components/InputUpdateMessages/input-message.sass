.conversation-input__input
  height: 3rem
  width: 75.6vw
  border: 0.15rem solid #665C5C
  border-radius: 0.5rem