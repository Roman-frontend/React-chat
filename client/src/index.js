import React from 'react';
import { render } from 'react-dom';
import { compose, createStore, applyMiddleware } from 'redux';
import {
  ApolloClient,
  InMemoryCache,
  ApolloProvider,
  gql,
} from '@apollo/client';
import {
  reactiveVarName,
  reactiveVarToken,
  reactiveVarEmail,
  reactiveVarId,
  reactiveDirectMessages,
  reactiveVarChannels,
  reactiveActiveChannelId,
  reactiveActiveDirrectMessageId,
  reactiveOnlineMembers,
} from './components/GraphQL/reactiveVariables';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { rootReducer } from './redux/reducers/rootReducer.js';
import './i18n';
import App from './App.js';

const typeDefs = gql`
  extend type Query {
    loginId: ID
  }
`;

const cache = new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        directMessagesId() {
          return reactiveDirectMessages();
        },
        channels() {
          reactiveVarChannels();
          return reactiveVarChannels();
        },
        token() {
          return reactiveVarToken();
        },
        name() {
          return reactiveVarName();
        },
        email() {
          return reactiveVarEmail();
        },
        id() {
          return reactiveVarId();
        },
        usersOnline() {
          return reactiveOnlineMembers();
        },
        activeChannelId: {
          read() {
            return reactiveActiveChannelId();
          },
        },
        activeDirectMessageId: {
          read() {
            return reactiveActiveDirrectMessageId();
          },
        },
      },
    },
    DirectMessage: {
      fields: {
        addNew: {
          read(directMessages = "Не визначене Ім'я", { variables }) {
            return directMessages;
          },
        },
      },
    },
  },
});

const client = new ApolloClient({
  uri: 'http://localhost:5000/graphql',
  typeDefs,
  cache,
  resolvers: {},
});

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
);

//цей ApolloProvider - для - apollo-client
const app = (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <App />
    </Provider>
  </ApolloProvider>
);

render(app, document.getElementById('root'));
