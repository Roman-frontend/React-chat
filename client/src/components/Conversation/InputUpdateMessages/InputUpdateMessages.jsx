import React from 'react';
import {
  ThemeProvider,
  makeStyles,
  createMuiTheme,
} from '@material-ui/core/styles';
import BorderColorIcon from '@material-ui/icons/BorderColor';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import { useDispatch, useSelector } from 'react-redux';
import { connect } from 'react-redux';
import {
  postMessage,
  postMessageToDirectMsg,
  putMessage,
} from '../../../redux/actions/actions.js';
import './input-message.sass';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    flexGrow: 1,
    fontSize: '3rem',
    textAlign: 'right',
  },
  addPeoples: {
    padding: theme.spacing(1),
    textAlign: 'bottom',
  },
}));

const theme = createMuiTheme({
  palette: {
    color: '#115293',
  },
});

export const InputUpdateMessages = React.memo((props) => {
  const {
    closeBtnChangeMsg,
    setCloseBtnChangeMsg,
    closeBtnReplyMsg,
    setCloseBtnReplyMsg,
    inputRef,
  } = props;
  const classes = useStyles();
  const dispatch = useDispatch();
  const name = useSelector((state) => state.userData.name);
  const userId = useSelector((state) => state.userData._id);
  const token = useSelector((state) => state.token);
  const activeChannelId = useSelector((state) => state.activeChannelId);
  const activeDirectMessageId = useSelector(
    (state) => state.activeDirectMessageId
  );

  /*   useEffect(() => {
    document.addEventListener('click', hidePopup);
  }, []);

  function hidePopup() {
    setPopupMessage(null);
    document.removeEventListener('click', hidePopup);
  } */

  function inputUpdateMessages(event) {
    event.preventDefault();
    const inputValue = inputRef.current.children[1].children[0].value;

    if (!(inputValue.trim() === '')) {
      if (closeBtnChangeMsg) changeMessageText(inputValue);
      else if (closeBtnReplyMsg) messageInReply(inputValue);
      else {
        newMessage(inputValue);
      }
      inputRef.current.children[1].children[0].value = null;
    }
  }

  async function changeMessageText(inputValue) {
    /* let messageForEdit = []

    const updatedArrayMessages = reduxMessages.map(message => {
      if (message._id === closeBtnChangeMsg) {
        message.text = inputValue
        messageForEdit.push(message)
        return message
      } else return message
    })
    
    const resPut = await putMessage(messageForEdit[0], closeBtnChangeMsg, null, token) */
    setCloseBtnChangeMsg(null);
  }

  const messageInReply = (response) => {
    const chatId = activeChannelId ? activeChannelId : activeDirectMessageId;
    const replyMsg = {
      id: Date.now(),
      userId,
      username: name,
      text: closeBtnReplyMsg.text,
      createdAt: new Date().toLocaleString(),
      chatId,
      reply: response,
    };
    dispatchMessage(replyMsg, chatId);
    setCloseBtnReplyMsg(null);
  };

  function newMessage(textMessage) {
    const chatId = activeChannelId ? activeChannelId : activeDirectMessageId;
    const newMsg = {
      id: Date.now(),
      userId,
      username: name,
      text: textMessage,
      createdAt: new Date().toLocaleString(),
      chatId,
    };
    dispatchMessage(newMsg, chatId);
  }

  function dispatchMessage(message, chatId) {
    if (activeChannelId) {
      dispatch(postMessage(token, { userId, ...message }, chatId));
    } else if (activeDirectMessageId) {
      dispatch(postMessageToDirectMsg(token, { ...message }, chatId));
    }
  }

  return (
    <div className={classes.root} id='mainInput'>
      <Grid container spacing={1}>
        <Grid item xs={1}>
          <BorderColorIcon
            className={classes.addPeoples}
            style={{ fontSize: 40, top: '1rem' }}
          />
        </Grid>
        <Grid item xs={11}>
          <form
            className={classes.root}
            noValidate
            autoComplete='off'
            onSubmit={(event) => inputUpdateMessages(event)}
          >
            <ThemeProvider theme={theme}>
              <TextField
                style={{ width: '67vw' }}
                className={'conversation-input__input'}
                label='Enter text'
                id='mui-theme-provider-standard-input'
                ref={inputRef}
              />
            </ThemeProvider>
          </form>
        </Grid>
      </Grid>
    </div>
  );
});

const mapDispatchToProps = {
  postMessage,
  postMessageToDirectMsg,
  putMessage,
};

export default connect(null, mapDispatchToProps)(InputUpdateMessages);
