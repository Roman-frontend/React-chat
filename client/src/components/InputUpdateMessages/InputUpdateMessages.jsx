import React, {useLayoutEffect, useEffect} from 'react'
import {useAuthContext} from '../../context/AuthContext.js'
import {useMessagesContext} from '../../context/MessagesContext.js'
import {useServer} from '../../hooks/Server.js'
import './input-message.sass'

export default function InputUpdateMessages(props) {
  const {name, userId} = useAuthContext()
  const {messages, setMessages, inputRef, activeChannelId} = useMessagesContext()
  const {postData, putData} = useServer()
  const {activeMessage, setActiveMessage} = props

  const copyMessages = messages.slice(0, messages.length);
  let updatedArrayMessages = []

  useEffect(() => { inputRef.current.focus() }, [])

  function inputUpdateMessages(event) {
    if ((event.key === "Enter") && !(inputRef.current.value === "")) {
      if (activeMessage.change) changeMessageText()
      else if (activeMessage.reply) messageInReply(inputRef.current.value)
      else newMessage(inputRef.current.value)
      
      setMessages(updatedArrayMessages)   
      inputRef.current.value = null
    }
  }

  async function changeMessageText() {
    let putMessage = []

    updatedArrayMessages = messages.map(message => {
      if (message._id === activeMessage.change) {
        message.text = inputRef.current.value
        putMessage.push(message)
        return message
      } else return message
    })
    
    const resPut = await putData(putMessage[0], activeMessage.change)
    if (resPut.messages) setMessages(resPut.messages.reverse())
    const object = Object.assign({}, {...activeMessage}, {'change': null})
    setActiveMessage({...object})
  }

  const messageInReply = async response => {
    copyMessages.unshift({
      id: Date.now(),
      userId: userId,
      username: name, 
      text: activeMessage.reply.text, 
      createdAt: new Date().toLocaleString(), 
      channelId: activeChannelId,
      reply: response,
    },) 
   
    const resPost = postData("postMessage", activeChannelId, copyMessages[0])
    if (resPost.channelMessages) setMessages(resPost.channelMessages.reverse())
    updatedArrayMessages = copyMessages
    const object = Object.assign({}, {...activeMessage}, {reply: null})
    setActiveMessage({...object}) 
  }

  function newMessage(textMessage) {
    
    copyMessages.unshift({
      id: Date.now(),
      userId: userId,
      username: name, 
      text: textMessage, 
      createdAt: new Date().toLocaleString(),
      channelId: activeChannelId,
    }, )  

    updatedArrayMessages = copyMessages
    const resPost = postData("postMessage", activeChannelId, updatedArrayMessages[0])
    if (resPost) {
      if (resPost.channelMessages) setMessages(resPost.channelMessages.reverse())
    }
  }

  return (
    <input 
      type="text" 
      className="conversation-input__input" 
      placeholder="Enter Text" 
      ref={inputRef} 
      onKeyUp={event => inputUpdateMessages(event)}
    />
  )
}