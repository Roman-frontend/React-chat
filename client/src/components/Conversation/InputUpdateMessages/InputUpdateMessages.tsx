import React, { useMemo, memo, useState, useEffect } from "react";
import { makeStyles } from "@mui/styles";
import BorderColorIcon from "@mui/icons-material/BorderColor";
import TextField from "@mui/material/TextField";
import Grid from "@mui/material/Grid";
import { useMutation, useQuery, useReactiveVar } from "@apollo/client";
import { AUTH } from "../../../GraphQLApp/queryes";
import {
  CREATE_MESSAGE,
  CHANGE_MESSAGE,
  GET_MESSAGES,
} from "../ConversationGraphQL/queryes";
import { wsSend } from "../../../WebSocket/soket";
import { activeChatId } from "../../../GraphQLApp/reactiveVars";
import IMessage from "../Models/IMessage";

interface IProps {
  testData: string;
  changeMessageRef: null | React.MutableRefObject<IMessage | null>;
  closeBtnChangeMsg: boolean;
  setCloseBtnChangeMsg: React.Dispatch<React.SetStateAction<boolean>>;
  closeBtnReplyMsg: boolean;
  setCloseBtnReplyMsg: React.Dispatch<React.SetStateAction<boolean>>;
  inputRef: React.MutableRefObject<HTMLInputElement | null>;
  popupMessage: null | IMessage;
}

interface ICacheMessage {
  messages: {
    id: string;
    chatMessages: IMessage[] | [];
  };
}

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    flexGrow: 1,
    fontSize: "3rem",
    textAlign: "right",
  },
}));

export const InputUpdateMessages = memo((props: IProps) => {
  const {
    testData,
    changeMessageRef,
    closeBtnChangeMsg,
    setCloseBtnChangeMsg,
    closeBtnReplyMsg,
    setCloseBtnReplyMsg,
    inputRef,
    popupMessage,
  } = props;
  const classes = useStyles();
  const { data: auth } = useQuery(AUTH);
  const activeChannelId = useReactiveVar(activeChatId).activeChannelId;
  const activeDirectMessageId =
    useReactiveVar(activeChatId).activeDirectMessageId;
  const [inputText, setInputText] = useState<string>("");

  const chatType = useMemo(() => {
    return activeDirectMessageId
      ? "DirectMessage"
      : activeChannelId
      ? "Channel"
      : null;
  }, [activeChannelId, activeDirectMessageId]);

  const chatId = useMemo(() => {
    return activeDirectMessageId || activeChannelId || null;
  }, [activeChannelId, activeDirectMessageId]);

  const { loading, data: messages } = useQuery(GET_MESSAGES, {
    variables: {
      chatId: "6288671cb24f6a89e861b98d",
      chatType: "DirectMessage",
      userId: "6288661c22cf8e8950762e14",
    },
    onError(data) {
      console.log("error __", data);
    },
  });

  const [
    createMessage,
    { loading: loadingCreate, error: errorCreate, data: dataCreate },
  ] = useMutation(CREATE_MESSAGE, {
    update: (cache, { data }) => {
      const cacheMsg: ICacheMessage | null = cache.readQuery({
        query: GET_MESSAGES,
        variables: {
          chatId: chatId || "6288671cb24f6a89e861b98d",
          chatType: chatType || "DirectMessage",
          userId: auth?.id || "6288661c22cf8e8950762e14",
        },
      });
      if (cacheMsg && data?.message) {
        const chatMessages = cacheMsg?.messages?.chatMessages || [];
        const newMsg = data.message.create;

        cache.writeQuery({
          query: GET_MESSAGES,
          data: {
            messages: {
              ...cacheMsg.messages,
              chatMessages: [...chatMessages, newMsg],
            },
          },
        });
      }
    },
    onCompleted(data) {
      setInputText("bla la la");
      sendMessageToWS(data.message.create);
    },
  });

  const [changeMessage] = useMutation(CHANGE_MESSAGE, {
    update: (cache, { data: { message } }) => {
      const cacheMsg: ICacheMessage | null = cache.readQuery({
        query: GET_MESSAGES,
        variables: { chatId, chatType, userId: auth.id },
      });
      if (cacheMsg && message?.change) {
        const cacheMessages: IMessage[] | [] =
          cacheMsg?.messages?.chatMessages || [];
        const chatMessages = cacheMessages.map((msg) => {
          return msg.id === message.change.id ? message.change : msg;
        });
        cache.writeQuery({
          query: GET_MESSAGES,
          data: { messages: { ...cacheMsg.messages, chatMessages } },
        });
      }
    },
    onError(error) {
      console.log(`Помилка ${error}`);
    },
    onCompleted(data) {
      sendMessageToWS(data.message.change);
    },
  });

  function sendMessageToWS(data: IMessage) {
    wsSend({
      meta: "sendMessage",
      action: "change",
      room: data.chatId,
      message: data,
    });
  }

  function inputUpdateMessages(event: React.KeyboardEvent<HTMLInputElement>) {
    event.preventDefault();
    const value = inputRef?.current?.value || "";

    //event.shiftKey - містить значення true - коли користувач нажме деякі з клавіш утримуючи shift
    if (value.trim() !== "" && !event.shiftKey && event.key === "Enter") {
      if (closeBtnChangeMsg) changeMessageText(value);
      else if (closeBtnReplyMsg) messageInReply(value);
      else newMessage(value);
      if (inputRef?.current?.value) inputRef.current.value = "";
    }
  }

  async function changeMessageText(text: string) {
    const newMsg = { id: popupMessage?.id || "", text, chatType };
    changeMessage({
      variables: { input: newMsg },
      optimisticResponse: { message: { change: { ...popupMessage, text } } },
    });
    if (changeMessageRef?.current) changeMessageRef.current = null;
    setCloseBtnChangeMsg(false);
  }

  const messageInReply = (text: string) => {
    const replyMsg = {
      chatId,
      chatType,
      senderId: auth?.id || "6288661c22cf8e8950762e14",
      replyOn: popupMessage?.text || "",
      text,
    };
    createMessage({
      variables: replyMsg,
      optimisticResponse: {
        message: {
          create: {
            chatId,
            chatType,
            createdAt: Date.now(),
            status: "sent",
            id: Date.now(),
            replyOn: popupMessage?.text || "",
            text,
            updatedAt: "",
            senderId: auth?.id || "6288661c22cf8e8950762e14",
            __typename: "MessagePayload",
          },
        },
      },
    });
    setCloseBtnReplyMsg(false);
  };

  function newMessage(text: string) {
    const newMsg = {
      chatId: chatId || "6288671cb24f6a89e861b98d",
      chatType: chatType || "DirectMessage",
      senderId: auth?.id || "6288661c22cf8e8950762e14",
      text,
    };
    createMessage({
      variables: newMsg,
      optimisticResponse: {
        message: {
          create: {
            chatId,
            chatType,
            createdAt: Date.now(),
            status: "sent",
            id: Date.now(),
            replyOn: null,
            text,
            updatedAt: "",
            senderId: auth?.id || "6288661c22cf8e8950762e14",
            __typename: "MessagePayload",
          },
        },
      },
    });
  }

  if (loading) {
    return (
      <>
        <p>loading</p>
      </>
    );
  }

  function changeInput(event: any) {
    if (inputRef?.current?.value) inputRef.current.value = event.target.value;
    setInputText(event.target.value);
  }

  return (
    <div className={classes.root} id="mainInput">
      {dataCreate && (
        <p data-testid="success-created-message">
          {dataCreate.message.create.text}
        </p>
      )}
      <Grid container spacing={1}>
        <Grid item xs={1}>
          <BorderColorIcon
            //color='input'
            style={{ fontSize: 40, top: "1rem", textAlign: "center" }}
          />
        </Grid>
        <Grid item xs={11}>
          <TextField
            inputProps={{ "data-testid": "on-key-up-main-input" }}
            //color='input'
            value={inputText}
            label="Enter text"
            variant="standard"
            inputRef={inputRef}
            autoFocus={true}
            onChange={(event) => changeInput(event)}
            onKeyUp={(event: React.KeyboardEvent<HTMLInputElement>) =>
              inputUpdateMessages(event)
            }
            sx={{
              paddingRight: "6vw",
              width: "-webkit-fill-available",
            }}
          />
        </Grid>
      </Grid>
    </div>
  );
});
