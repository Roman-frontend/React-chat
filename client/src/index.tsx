import React from 'react';
import { render } from 'react-dom';
import { ApolloProvider } from '@apollo/client';
import './i18n';
import App from './App';
import { client } from './GraphQLApp/apolloClient';
import ErrorBoundary from './components/Helpers/ErrorBoundare';
import { SnackbarProvider } from 'notistack';

//цей ApolloProvider - для - apollo-client
const app = (
  <React.StrictMode>
    <ApolloProvider client={client}>
      <ErrorBoundary>
        <SnackbarProvider
          maxSnack={3}
          anchorOrigin={{
            vertical: 'top',
            horizontal: 'right',
          }}
        >
          <App />
        </SnackbarProvider>
      </ErrorBoundary>
    </ApolloProvider>
  </React.StrictMode>
);
render(app, document.getElementById('root'));
