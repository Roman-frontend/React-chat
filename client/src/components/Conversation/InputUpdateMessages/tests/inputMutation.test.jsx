import React from "react";
import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { act } from "react-dom/test-utils";
// fireEvent vs userEvent - https://www.youtube.com/watch?v=3YrxcAXkUKQ
import userEvent from "@testing-library/user-event";
import { SnackbarProvider } from "notistack";
import TestRenderer from "react-test-renderer";
// https://www.apollographql.com/docs/react/development-testing/testing
import { MockedProvider } from "@apollo/client/testing";
import mocks from "./mocks";
import { InputUpdateMessages } from "../InputUpdateMessages";
import "@testing-library/jest-dom";
import "@testing-library/dom";

function sleep(ms) {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

describe("InputUpdateMessages graphql mutation", () => {
  test("create message", async () => {
    try {
      await act(async () =>
        render(
          <MockedProvider mocks={mocks} addTypename={false}>
            <InputUpdateMessages inputRef={{ current: null }} />
          </MockedProvider>
        )
      );

      const component = await TestRenderer.create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <InputUpdateMessages />
        </MockedProvider>
      );

      // const inputForKeyUp = await waitFor(() =>
      //   screen.getByTestId("on-key-up-main-input").querySelector("input")
      // );
      const inputForKeyUp = await waitFor(() =>
        screen.getByTestId("on-key-up-main-input")
      );
      expect(inputForKeyUp).toBeInTheDocument();
      fireEvent.change(inputForKeyUp, { target: { value: "222" } });
      // expect(inputForKeyUp).toHaveValue("222");
      expect(inputForKeyUp.value).toBe("222");
      expect(inputForKeyUp).toHaveAttribute("value", "222");
      expect(inputForKeyUp).toMatchSnapshot();
      await fireEvent.keyUp(inputForKeyUp, { key: "Enter" });
      const successNotificationAfterCreate = await waitFor(() =>
        screen.getByTestId("success-created-message")
      );
      expect(successNotificationAfterCreate).toBeInTheDocument();
      // userEvent.keyUp(inputForKeyUp, "Enter");
      // userEvent.clear(inputForKeyUp);

      // await waitFor(() => screen.debug());
    } catch (error) {
      console.log(error);
    }
  });
});
