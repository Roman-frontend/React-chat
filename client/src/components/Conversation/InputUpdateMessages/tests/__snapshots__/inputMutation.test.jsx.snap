// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputUpdateMessages graphql mutation create message 1`] = `
<input
  aria-invalid="false"
  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
  data-testid="on-key-up-main-input"
  id=":r0:"
  type="text"
  value="222"
/>
`;
